Bio/SeqIO/XdnaIO.py

```diff
@@ -0,0 +1,296 @@
+        _read_overhang(handle)  # right-side overhang
+        _read_overhang(handle)  # left-side overhang
+
+        # Read the features
+        num_features = unpack('>B', _read(handle, 1))[0]
+        while num_features > 0:
+            _read_feature(handle, record)
+            num_features -= 1
+
+    yield record
+
+
+class XdnaWriter(SequenceWriter):
+    """Write files in the Xdna format."""
+
+    def write_file(self, records):
+        """Write the specified record to a Xdna file.
+
+        Note that the function expects a list of records as per the
+        SequenceWriter interface, but the list should contain only one
+        record as the Xdna format is a mono-record format.
+        """
+        if not records:
+            raise ValueError("Must have one sequence")
+        if len(records) > 1:
+            raise ValueError("More than one sequence found")
+
+        record = records[0]
+
+        alptype = type(record.seq.alphabet)
+        if issubclass(alptype, Alphabet.DNAAlphabet):
+            seqtype = 1
+        elif issubclass(alptype, Alphabet.RNAAlphabet):
+            seqtype = 3
+        elif issubclass(alptype, Alphabet.ProteinAlphabet):
+            seqtype = 4
+        else:
+            seqtype = 0
+
+        if record.annotations.get('topology', 'linear') == 'circular':
+            topology = 1
+        else:
+            topology = 0
+
+        # We store the record's id and description in the comment field.
+        # Make sure to avoid duplicating the id if it is already
+        # contained in the description.
+        if record.description.startswith(record.id):
+            comment = record.description
+        else:
+            comment = '{} {}'.format(record.id, record.description)
+
+        # Write header
+        self.handle.write(pack('>BBB25xII60xI11xB',
+                               0,   # version
+                               seqtype, topology, len(record),
+                               0,   # negative length
+                               len(comment),
+                               255  # end of header
+                               ))
+
+        # Actual sequence and comment
+        self.handle.write(str(record.seq).encode('ASCII'))
+        self.handle.write(comment.encode('ASCII'))
+
+        self.handle.write(pack('>B', 0))    # Annotation section marker
+        self._write_pstring('0')            # right-side overhang
+        self._write_pstring('0')            # left-side overhand
+
+        # Write features
+        # We must skip features with fuzzy locations as they cannot be
+        # represented in the Xdna format
+        features = [f for f in record.features if type(f.location.start) == ExactPosition and type(f.location.end) == ExactPosition]
+        # We also cannot store more than 255 features as the number of
+        # features is stored on a single byte...
+        if len(features) > 255:
+            features = features[:255]
```

Comment:
I think that data loss deserves a warning (as done in some of the other writers)